### PR TITLE
fix(ivy): ensure `DebugNode`/`DebugElement` are tree-shakeable in Ivy

### PR DESCRIPTION
**Pull request by gkalpak**
There are different `DebugNode`/`DebugElement` implementations (and associated helper functions) for ViewEngine and Ivy. Additionally, these classes/functions, which are defined inside the `core` package, are imported by the `platform-browser` package.

Previously, this code was not tree-shaken as expected in Ivy. #30130 partially addressed the issue, but only for the case where `core` and `platform-browser` end up in the same closure after webpack's scope hoisting. In cases where this is not the case, our webpack/terser based tooling is not capable of tree-shaking it.

This commit fixes the problem, by ensuring that the code retained in Ivy mode (due to the cross-package import) does not unnecessarily reference `DebugNode`/`DebugElement`, allowing the code to be tree-shaken away. This results in a 7.6KB reduction in the size of the main angular.io
bundle.

Jira issue: [FW-1802](https://angular-team.atlassian.net/browse/FW-1802)

### CODE REVIEW

File: packages/core/src/debug/debug_node.ts [hunk 1]
```diff
@@ -727,6 +727,18 @@ export function getDebugNode__POST_R3__(nativeNode: any): DebugNode|null {
...
+
+export function getDebugNodeR2__PRE_R3__(nativeNode: any): DebugNode|null {
+  return getDebugNode__PRE_R3__(nativeNode);
+}
+
+export function getDebugNodeR2__POST_R3__(_nativeNode: any): DebugNode|null {
+  return null;
+}
+
+export const getDebugNodeR2: (nativeNode: any) => DebugNode | null = getDebugNodeR2__PRE_R3__;
```
Comment (gkalpak) [inline packages/core/src/debug/debug_node.ts]: Not sure what would be an appropriate name :innocent:

File: aio/scripts/_payload-limits.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "master": {
       "uncompressed": {
         "runtime-es2015": 2987,
-        "main-es2015": 456604,
+        "main-es2015": 448956,
```
Comment (IgorMinar) [inline aio/scripts/_payload-limits.json]: woot!!!

File: integration/_payload-limits.json [hunk 3]
```diff
@@ -39,7 +39,7 @@
     "master": {
       "uncompressed": {
         "runtime-es2015": 2289,
-        "main-es2015": 254657,
+        "main-es2015": 247225,
```
Comment (IgorMinar) [inline integration/_payload-limits.json]: sweet!

File: packages/core/src/debug/debug_node.ts (hunk 1 above)
Comment (IgorMinar) [reply packages/core/src/debug/debug_node.ts]: lgtm

**Review by IgorMinar — APPROVED** [review]
George, you rock!

thanks for figuring this out.

❤️

**Pull request closed without merging**